frontend/packages/noobaa-storage-plugin/src/components/bucket-class/wizard-pages/backingstore-page.tsx

```diff
@@ -0,0 +1,255 @@
+import * as classNames from 'classnames';
+import * as fuzzy from 'fuzzysearch';
+import * as _ from 'lodash';
+import * as React from 'react';
+import { Form, FormGroup, TextInput, Title } from '@patternfly/react-core';
+import {
+  IRowData,
+  Table,
+  TableBody,
+  TableGridBreakpoint,
+  TableHeader,
+  TableVariant,
+} from '@patternfly/react-table';
+import { Firehose, FirehoseResult, ResourceLink } from '@console/internal/components/utils';
+import { K8sResourceKind, referenceForModel } from '@console/internal/module/k8s';
+import { connectToModel } from '@console/internal/kinds';
+import { NooBaaBackingStoreModel } from '@console/noobaa-storage-plugin/src/models';
+import { Action, BackingStoreStateType, State } from '../state';
+import { HelpTextCreator } from './help-text';
+
+const tableColumnClasses = [
+  classNames('col-md-4', 'col-sm-4', 'col-xs-6', 'nb-bc-bs-table__data'),
+  classNames('col-md-3', 'col-sm-4', 'col-xs-5', 'nb-bc-bs-table__data'),
+  classNames('col-md-2', 'col-sm-3', 'hidden-xs'),
+  classNames('col-md-2', 'hidden-sm', 'hidden-xs'),
+];
+
+const nameMap = Object.freeze({
+  'aws-s3': 'awsS3',
+});
+
+const columns = [
+  {
+    title: 'Name',
+    props: { className: tableColumnClasses[0] },
+  },
+  {
+    title: 'BucketName',
+    props: { className: tableColumnClasses[1] },
+  },
+  {
+    title: 'Type',
+    props: { className: tableColumnClasses[2] },
+  },
+  {
+    title: 'Region',
+    props: { className: tableColumnClasses[3] },
+  },
+];
+
+const filterSelected = (list: BackingStoreStateType[], tableId: number) => {
+  const sort = (x, y) => x.id.localeCompare(y.id);
+  return list.filter((e) => e.selectedBy === tableId || e.selectedBy === '').sort(sort);
```

Comment:
can you reduce the complexity of this statement.

Reply:
tried it. couldn't come up with anything.